R/create_monthly_costs.R

```diff
@@ -19,16 +20,57 @@ create_monthly_costs <- function(data,
   costs <- data %>%
     dplyr::select(dplyr::ends_with("_beddays")) %>%
     dplyr::rename_with(~ stringr::str_replace(., "_beddays", "_cost"))
+  # Fix the instances where the episode is a daycase;
+  # these will sometimes have 0.33 for the yearstay,
+  # this should be applied to the relevant month.
+  costs_daycase <- data %>%
+    dplyr::select(-dplyr::ends_with("_beddays")) %>%
+    dplyr::mutate(
+      daycase_added = (.data$record_keydate1 == .data$record_keydate2)
+    ) %>%
+    dplyr::mutate(daycase_check = .data$daycase_added) %>%
+    dplyr::mutate(whichmonth = dplyr::if_else(
+      .data$daycase_added,
+      lubridate::month(data$record_keydate1),
+      NA
+    )) %>%
+    dplyr::mutate(
+      whichmonth = month.abb[.data$whichmonth] %>%
+        tolower() %>%
+        paste0("_cost"),
+      daycase_added = dplyr::if_else(.data$daycase_added, 1, 0)
+    ) %>%
```

Comment:
Can trim this code a bit by getting lubridate to give the abbreviated date label
```suggestion
    dplyr::mutate(whichmonth = dplyr::if_else(
      .data$daycase_added,
      lubridate::month(data$record_keydate1, label = TRUE) %>%
        tolower() %>%
        paste0("_cost"),
      NA
    )) %>%
    dplyr::mutate(
      daycase_added = dplyr::if_else(.data$daycase_added, 1, 0)
    ) %>%
```